lib/ui/widgets/cached_image.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:flutter_cache_manager/flutter_cache_manager.dart';
+import 'package:cached_network_image/cached_network_image.dart';
+import 'package:flutter/widgets.dart';
+import 'package:reaxit/config.dart' as config;
+
+/// A [BaseCacheManager] with customized configurations.
+class _ThaliaCacheManager extends CacheManager with ImageCacheManager {
+  static const key = 'thaliaCachedDate';
+
+  static final _ThaliaCacheManager _instance = _ThaliaCacheManager._();
+  factory _ThaliaCacheManager() => _instance;
+
+  _ThaliaCacheManager._()
+      : super(Config(
+          key,
+          stalePeriod: config.cacheStalePeriod,
+          maxNrOfCacheObjects: config.cacheMaxObjects,
+        ));
+}
+
+/// Wrapper for [CachedNetworkImage] with sensible defaults.
+class CachedImage extends CachedNetworkImage {
+  CachedImage({
+    required String imageUrl,
+    BoxFit fit = BoxFit.cover,
+    Duration fadeOutDuration = const Duration(milliseconds: 200),
+    Duration fadeInDuration = const Duration(milliseconds: 200),
+    required String placeholder,
+  }) : super(
+          imageUrl: imageUrl,
+          cacheManager: _ThaliaCacheManager(),
+          cacheKey: Uri.parse(imageUrl).replace(query: '').toString(),
```

Comment:
Aren't you supposed to cache images with the querystring? I know this makes it hard to cache the photos then.
Only thing I know is that in a browser you usually append a different querystring when your CSS/JS version changes.

Reply:
I'm not sure I understand what you mean. What this does right now is:
1. Check in the cache whether there is an entry labelled as the image url without query part (so only the identifier, not the signature)
2. If not: request the image url including the query part, and add it to the cache with the label without the signature.

I have to say I haven't dived very deep into how the cache manager package exactly handles the requests. I believe this approach to caching is much more basic than what a browser would normally do for you. Iirc there is some mention of using caching headers, but I suspect that is very limited. The core idea is that there is just a local database of files.

The most basic approach (as documented in the flutter documentation) is to just leave out the cacheKey entirely. It then defaults to the full url, which means that when the signature changes, the entry with the old signature is not used, and that sounds quite wasteful in our case (normally authentication is done with headers, not a query).

Reply:
I meant that usually you use the full urls like that default that you mention.

Was just guessing this was intended and because of the signature. So I have my answer :)

Reply:
Exactly, this wouldn't work if the query is used for anything other than authentication, but they aren't here. I'll add some documentation of this.